### PR TITLE
Refactor Secure Ingress tasks

### PR DESCRIPTION
**Pull request by howardjohn**
Ingress SDS is now the default and recommended. This removes the file
mount task, which is no longer recommended and will avoid confusion, and
cleans up the task a bit. I also documented the new supported secret
formats.

### CODE REVIEW

**Comment by howardjohn** [comment]
Fixes https://github.com/istio/istio/issues/21076

**Comment by howardjohn** [comment]
Need to fix up some links but should be ready for review

**Comment by JimmyCYJ** [comment]
Thanks for the work! I am working on a doc about migrating file mount based approach to SDS based approach with @myidpt.

**Comment by howardjohn** [comment]
@JimmyCYJ does the work here conflict with that work? Or will that be in addition to this

File: content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md [hunk 1]
```diff
@@ -58,95 +50,14 @@ For this task you can use your favorite tool to generate certificates and keys.
...
-        spec:
-          containers:
-          - image: docker.io/citizenstig/httpbin
-            imagePullPolicy: IfNotPresent
-            name: httpbin
-            ports:
-            - containerPort: 8000
-    EOF
-    {{< /text >}}
+1.  Ensure you have deployed the [httpbin]({{< github_tree >}}/samples/httpbin) service.
```
Comment (adammil2000) [inline content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md:55]: Can you add a link here to the step where this occurs?

File: content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md [hunk 2]
```diff
@@ -510,21 +420,11 @@ $ kubectl create -n istio-system secret generic httpbin-credential --from-file=k
 
     {{< /text >}}
 
-1. Instead of creating a `httpbin-credential` secret to hold all the credentials, you can
-   create two separate secrets:
+Istio supports reading a few different formats, to support integration with various tools such as [cert-manager](/docs/ops/integrations/certmanager/):
```
Comment (adammil2000) [inline content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md:423]: Istio supports several formats of what? I assume a config file or secrets file?

Comment (howardjohn) [reply content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md:423]: Yeah this would be the Secret object. So Ill reword this to `Istio supports reading a few different formats of Secrets ....`

**Comment by JimmyCYJ** [comment]
> @JimmyCYJ does the work here conflict with that work? Or will that be in addition to this

Thanks for asking. I notice that you have removed this guide for file-mount content/en/docs/tasks/traffic-management/ingress/secure-ingress-mount/index.md.

My doc references to configurations in this doc, and lists steps about migrations:
single TLS gateway (file mount->SDS)
single mTLS gateway (file mount->SDS)
multiple host gateway (file mount->SDS)

I think I can replace the links to file mount guide in my doc with some short descriptions.

File: content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md [hunk 3]
```diff
@@ -510,21 +420,11 @@ $ kubectl create -n istio-system secret generic httpbin-credential --from-file=k
...
-    Create the two separate secrets with the following commands:
-
-    {{< text bash >}}
-    $ kubectl -n istio-system delete secret httpbin-credential
-    $ kubectl create -n istio-system secret generic httpbin-credential  \
-    --from-file=key=httpbin.example.com.key --from-file=cert=httpbin.example.com.crt
-    $ kubectl create -n istio-system secret generic httpbin-credential-cacert  \
-    --from-file=cacert=example.com.crt
-    {{< /text >}}
+* A Secret with keys `tls.key` and `tls.crt`, as described above. For mutual TLS, a `ca.crt` key can be used.
```
Comment (JimmyCYJ) [inline content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md:425]: Could you use "A TLS secret with keys ... " and A generic secret with keys ... "?

**Comment by myidpt** [comment]
@howardjohn Thanks for the work. I think we need to come up with a deprecation plan for the file-mount based ingress gateway. Removing the doc is fine but we are still supporting the legacy users.
@JimmyCYJ can write a task for the migration. He already has a doc about it.

**Comment by howardjohn** [comment]
@myidpt yes I don't intend to remove support for this, just don't want it in the mainline docs to confuse users. Advanced users can look at the gateway reference docs/the migration guide I think

File: content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md (hunk 3 above)
Comment (howardjohn) [reply content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md:425]: done

**Comment by howardjohn** [comment]
@adammil2000 existing comments are addressed if you have a chance to review again. Thanks!

File: content/en/news/releases/1.2.x/announcing-1.2/change-notes/index.md [hunk 4]
```diff
@@ -30,7 +30,7 @@ aliases:
 - **Fixed** [SAN](https://tools.ietf.org/html/rfc5280#section-4.2.1.6) field in workload certificates is set as `critical`. This fixes the issue that some custom certificate verifiers cannot verify Istio certificates.
 - **Fixed** mutual TLS probe rewrite for HTTPS probes.
 - **Graduated** [SNI with multiple certificates support at ingress gateway](/docs/reference/config/networking/gateway/) from Alpha to Stable.
-- **Graduated** [certification management on Ingress Gateway](/docs/tasks/traffic-management/ingress/secure-ingress-sds/) from Alpha to Beta.
+- **Graduated** [certification management on Ingress Gateway](/docs/tasks/traffic-management/ingress/secure-ingress/) from Alpha to Beta.
```
Comment (frankbu) [inline content/en/news/releases/1.2.x/announcing-1.2/change-notes/index.md:33]: I think you should point this to the archive as described here: https://preliminary.istio.io/about/contribute/remove-content/#reconcile-broken-links

File: content/en/news/releases/1.3.x/announcing-1.3/change-notes/index.md [hunk 5]
```diff
@@ -32,7 +32,7 @@ aliases:
 - **Added** support for [introspection](/docs/ops/diagnostic-tools/controlz/) to Citadel.
 - **Added** metrics to the `/metrics` endpoint of Citadel Agent on port 15014 to monitor the SDS service.
 - **Added** diagnostics to the Citadel Agent using the `/debug/sds/workload` and `/debug/sds/gateway` on port 8080.
-- **Improved** the ingress gateway to [load the trusted CA certificate from a separate secret](/docs/tasks/traffic-management/ingress/secure-ingress-sds/#configure-a-mutual-tls-ingress-gateway) when using SDS.
+- **Improved** the ingress gateway to [load the trusted CA certificate from a separate secret](/docs/tasks/traffic-management/ingress/secure-ingress/) when using SDS.
```
Comment (frankbu) [inline content/en/news/releases/1.3.x/announcing-1.3/change-notes/index.md:35]: same here, point to archive doc.

File: content/en/news/releases/1.2.x/announcing-1.2/change-notes/index.md (hunk 4 above)
Comment (howardjohn) [reply content/en/news/releases/1.2.x/announcing-1.2/change-notes/index.md:33]: I am not against it but according to (my interpretation of) the flow chart this seems correct. There is a new document available (this one) so it says to link to the new doc

Comment (frankbu) [reply content/en/news/releases/1.2.x/announcing-1.2/change-notes/index.md:33]: That's step 2 of the flowchart. The first step is to check if the link is from an announcing-x.x doc and if YES, to point to the corresponding archive version

Comment (howardjohn) [reply content/en/news/releases/1.2.x/announcing-1.2/change-notes/index.md:33]: Oops. thanks!
